### PR TITLE
Add the swish function

### PR DESCRIPTION
**Pull request by Shashi456**
Can someone check the vjp for the function?

Inspired from [here](https://github.com/tensorflow/tensorflow/blob/7d9144dc5f7725f9b4fbf005234d95fa7ded4baa/tensorflow/python/ops/nn_impl.py#L521).

### CODE REVIEW

File: Tests/TensorFlowTests/OperatorTests/MathTests.swift [hunk 1]
```diff
@@ -250,6 +250,13 @@ final class MathOperatorTests: XCTestCase {
         assertEqual(y, expectedY, accuracy: 1e-5)
     }
 
+    func testSwish() {
```
Comment (awav) [inline Tests/TensorFlowTests/OperatorTests/MathTests.swift]: Looks good! Could you test the gradient as well? Thanks!

Comment (Shashi456) [reply Tests/TensorFlowTests/OperatorTests/MathTests.swift]: I have another PR testing gradients for a lot of these functions, I'll have that up in a few days.

File: Sources/TensorFlow/Operators/Math.swift [hunk 2]
```diff
@@ -1330,6 +1330,28 @@ func _vjpSelu<T: TensorFlowFloatingPoint>(
...
+    x * sigmoid(x)
+}
+
+@inlinable
+@derivative(of: swish)
+func _vjpSwish<T: TensorFlowFloatingPoint>(
+    _ x: Tensor<T>
+) -> (value: Tensor<T>, pullback: (Tensor<T>) -> Tensor<T>) {
+    return (swish(x), { v in 
+        let grad = sigmoid(x) * (1.0 + x * (1 - sigmoid(x)))
```
Comment (awav) [inline Sources/TensorFlow/Operators/Math.swift]:
```suggestion
        let sigmoid_features = sigmoid(x)
        let grad = sigmoid_features * (1.0 + x * (1 - sigmoid_features))
```

Comment (awav) [reply Sources/TensorFlow/Operators/Math.swift]: I think it does make sense to compute it explicitly once when x tensor is large.

File: Sources/TensorFlow/Operators/Math.swift [hunk 3]
```diff
@@ -1330,6 +1330,28 @@ func _vjpSelu<T: TensorFlowFloatingPoint>(
...
+/// https://arxiv.org/abs/1710.05941
+@inlinable
+@differentiable
+public func swish<T: TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
+    x * sigmoid(x)
+}
+
+@inlinable
+@derivative(of: swish)
+func _vjpSwish<T: TensorFlowFloatingPoint>(
```
Comment (marcrasi) [inline Sources/TensorFlow/Operators/Math.swift]: We shouldn't need a custom vjp at all here. The compiler should be able to synthesize this automatically! Can you remove it?

File: Tests/TensorFlowTests/OperatorTests/MathTests.swift (hunk 1 above)
Comment (marcrasi) [reply Tests/TensorFlowTests/OperatorTests/MathTests.swift]: Once we have these tests, they'll verify that the compiler indeed does compute a correct vjp for the function.

File: Sources/TensorFlow/Operators/Math.swift (hunk 3 above)
Comment (marcrasi) [reply Sources/TensorFlow/Operators/Math.swift]: Ah, I see a justification for the custom vjp at [the link you posted](https://github.com/tensorflow/tensorflow/blob/7d9144dc5f7725f9b4fbf005234d95fa7ded4baa/tensorflow/python/ops/nn_impl.py#L521).

That's a good justification, and this custom vjp does do the same thing.

You should add a comment explaining why there is a custom vjp.

**Pull request merged**